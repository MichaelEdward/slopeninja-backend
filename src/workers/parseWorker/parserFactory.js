import cheerio from 'cheerio';
import { AllHtmlEntities } from 'html-entities';

const htmlEntities = new AllHtmlEntities();

export const removeBackSlashes = (text) => {
  const decoded = text ? text.toString().replace(/\\/g, '') : null;
  return decoded;
}

export const xyz = (text) => {
  const decoded = text ? text.toString().replace(/\u0026amp;/gu, '__JAKE__') : null;
  return decoded;
}

export const decodeEntities = (text) => {
  // best way I found to normalize unicode in input
  const unicodeInput = JSON.stringify(JSON.parse(text.toString()));
  return htmlEntities.decode(unicodeInput);
}

export const createHtmlParser = (key, parser, preParser = t => t) => async (html) => {
  const $ = cheerio.load(preParser(html), { decodeEntities: true });

  const keyData = await parser($);

  return {
    [key]: keyData
  };
}

export const createJSONParser = (key, parser, preParser = t => t) => async (data = "{}") => {
  let json = JSON.parse(preParser(data));

  let d;
  if (json) {
    d = json;
  }

  const keyData = await parser(d);

  return {
    [key]: keyData
  };
}
