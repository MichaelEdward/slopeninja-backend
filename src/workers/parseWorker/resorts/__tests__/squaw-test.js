import fs from 'fs';
import {
  parseSquawSnow,
  parseSquawLifts,
  parseSquawLiftList,
  parseSquawTrailList,
  parseSquawTrails,
} from '../squaw';
import { createHtmlParser } from '../../parserFactory';

test('fetches Squaw snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/squaw-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseSquawSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: 'sunny',
      temprature: 52,
      baseCondition: null,
      newSnow: 0,
      snowDepthBase: null,
      snowDepthSummit: 161,
    },
  });
})

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseSquawSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temprature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    }
  });
});

test('fetches Squaw lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('lifts', parseSquawLifts)(htmlText);
  expect(resortData).toEqual({
    lifts: {
      total: null,
      open: 1,
    }
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('lifts', parseSquawLifts)('<html></html>');
  expect(resortData).toEqual({
    lifts: {
      total: null,
      open: null,
    }
  });
});

test('fetches Squaw lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('liftList', parseSquawLiftList)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('liftList', parseSquawLiftList)('<html></html>');
  expect(resortData).toMatchObject({ liftList: [] });
});

test('fetches Squaw trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('trailList', parseSquawTrailList)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trailList', parseSquawTrailList)('<html></html>');
  expect(resortData).toMatchObject({ trailList: [] });
});


test('fetches Squaw trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('trails', parseSquawTrails)(htmlText);
  expect(resortData).toEqual({
    trails: {
      total: null,
      open: 0,
    }
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trails', parseSquawTrails)('<html></html>');
  expect(resortData).toEqual({
    trails: {
      total: null,
      open: null,
    }
  });
});
