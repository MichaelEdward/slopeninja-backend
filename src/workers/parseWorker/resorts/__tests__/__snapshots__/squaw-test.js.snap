// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Squaw lift list correctly 1`] = `
Object {
  "liftList": Array [
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Aerial Tram",
      "status": "open",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Funitel",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Exhibition",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Far East Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "First Venture",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "KT22 Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Olympic Lady",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Red Dog",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Squaw Creek",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Squaw One Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Boon",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Murphy",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "SnoVentures Carpet",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Tucker",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Lower Mountain",
      "name": "Wylee",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Bailey's Beach",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Belmont",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Big Blue Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Broken Arrow Lift",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Emigrant",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Gold Coast Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Granite Chief Lift",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Headwall Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Mountain Meadow",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Shirley Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Siberia Express",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Silverado",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "Solitude",
      "status": "closed",
    },
    Object {
      "category": "Lifts-Upper Mountain",
      "name": "The Pulley",
      "status": "closed",
    },
    Object {
      "category": "SnoVentures",
      "name": "Mini-snowmobile",
      "status": "closed",
    },
    Object {
      "category": "SnoVentures",
      "name": "Snow Tubing",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches Squaw trail list correctly 1`] = `
Object {
  "trailList": Array [
    Object {
      "category": "Broken Arrow",
      "level": "advanced",
      "name": "Broken Arrow",
      "status": "closed",
    },
    Object {
      "category": "Broken Arrow",
      "level": "advanced",
      "name": "Powerline Face",
      "status": "closed",
    },
    Object {
      "category": "Broken Arrow",
      "level": "advanced",
      "name": "Swale",
      "status": "closed",
    },
    Object {
      "category": "Broken Arrow",
      "level": "advanced",
      "name": "Tower 16",
      "status": "closed",
    },
    Object {
      "category": "Exhibition",
      "level": "intermediate",
      "name": "Easy Street",
      "status": "closed",
    },
    Object {
      "category": "Exhibition",
      "level": "intermediate",
      "name": "Julia's Gold",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "beginner",
      "name": "Eagle Pass",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "beginner",
      "name": "Ramp Run",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "beginner",
      "name": "Riviera",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "beginner",
      "name": "Snow Flower",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "beginner",
      "name": "To Shirley/High Camp",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "intermediate",
      "name": "Emigrant",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "intermediate",
      "name": "Emigrant Face",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "intermediate",
      "name": "Gold Coast Face",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "intermediate",
      "name": "Monument Ridge",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "intermediate",
      "name": "Mystery",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "advanced",
      "name": "Funnel",
      "status": "closed",
    },
    Object {
      "category": "Goldcoast/Emigrant",
      "level": "advanced",
      "name": "Mainline Pocket",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "The Attic",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Arete",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Bottleneck Gully",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Break It Out",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Granite Alley",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Granite Chief Peak",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Granite Glades",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Hidden Bowl",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "High Voltage",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Magoos",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Main Backside",
      "status": "closed",
    },
    Object {
      "category": "Granite Chief",
      "level": "advanced",
      "name": "Shirley Access",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "intermediate",
      "name": "Newport",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Bullet",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Classic",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Cornice Bowl",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Garbage Chutes",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Headwall Face",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Hogsback",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Horse Trails",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Hourglass",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Norm's",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "North Bowl",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "Sun Bowl",
      "status": "closed",
    },
    Object {
      "category": "Headwall",
      "level": "advanced",
      "name": "The Slot",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Larkspur",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Lupine",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Monkey Flower",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Mule's Ear",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Poppy",
      "status": "closed",
    },
    Object {
      "category": "High Camp Beginner",
      "level": "beginner",
      "name": "Shooting Star",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Saddle",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Chute 75",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Dead Tree",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Enchanted Forest",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "GS Bowl",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Heidi's",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Heidi's Glades",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "KT Backcountry Gate",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Moseley's",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Nose",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Red Dog Ridge",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Rock Garden",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Saddle Face",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Schimmelpfennig Bowl",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Strawberry Fields",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Tamara's",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "The Fingers",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "W Face Alternates",
      "status": "closed",
    },
    Object {
      "category": "KT-22",
      "level": "advanced",
      "name": "Women's Downhill",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "beginner",
      "name": "Easy Slider",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Home Run",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Juniper Spire",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Mountain Run",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Olympic High",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Packers",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "intermediate",
      "name": "Sunnyside",
      "status": "closed",
    },
    Object {
      "category": "Mountain Run",
      "level": "advanced",
      "name": "Spring Bowl",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "intermediate",
      "name": "Champs Elysees",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Christmas Tree",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Cushman's",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Dog Leg",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Lower Dog Leg",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Lwr Champs Elysees",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Poulsens Gully",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Red Dog Face",
      "status": "closed",
    },
    Object {
      "category": "Red Dog",
      "level": "advanced",
      "name": "Upper Dog Leg",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Atkinson's 3rd",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Freeway",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Hill's 2nd",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Marrilac's 5th",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Rainbow Bowl",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Shirley Bowl",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Standteiner's 4th",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "To Solitude",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "intermediate",
      "name": "Tomlinson's 1st",
      "status": "closed",
    },
    Object {
      "category": "Shirley/Solitude",
      "level": "advanced",
      "name": "Reuter's 6th",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "intermediate",
      "name": "Racers",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "intermediate",
      "name": "Siberia Ridge Run",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "intermediate",
      "name": "Siberia Run",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "intermediate",
      "name": "Yellow Trail",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Chimney",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Extra",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Main",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "National",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Red Trail",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Siberia Bowl",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Siberia Ridge",
      "status": "closed",
    },
    Object {
      "category": "Siberia",
      "level": "advanced",
      "name": "Upper Sun Bowl",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Bailey's Cirque",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Billy's",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Far Side",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Fatal Attraction",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 1",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 2",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 3",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 4",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 5",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 6",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 7",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Gate 8",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Hanging Gardens",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Jake's",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Kathmandu",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Land Bridge",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Medusa",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Oregon Trail",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Trail 90",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Tram Bowl",
      "status": "closed",
    },
    Object {
      "category": "Silverado",
      "level": "advanced",
      "name": "Waimea",
      "status": "closed",
    },
    Object {
      "category": "SnoVentures",
      "level": "beginner",
      "name": "Pioneer",
      "status": "closed",
    },
    Object {
      "category": "SnoVentures",
      "level": "beginner",
      "name": "Start Me Up",
      "status": "closed",
    },
    Object {
      "category": "Squaw Creek",
      "level": "intermediate",
      "name": "Lake View",
      "status": "closed",
    },
    Object {
      "category": "Squaw Creek",
      "level": "intermediate",
      "name": "Valley View",
      "status": "closed",
    },
    Object {
      "category": "Squaw Creek",
      "level": "advanced",
      "name": "Knob Hill",
      "status": "closed",
    },
    Object {
      "category": "Squaw Creek",
      "level": "advanced",
      "name": "Montezumas",
      "status": "closed",
    },
    Object {
      "category": "Squaw Creek",
      "level": "advanced",
      "name": "Strainer",
      "status": "closed",
    },
  ],
}
`;
